mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -282,6 +272,23 @@ def settings_path(self):
             return self._old_settings_path
         return join(xdg.BaseDirectory.save_config_path(get_xdg_base(), 'skills', self.skill_id), 'settings.json')
 
+    @property
+    def settings(self):
+        if self._settings is not None:
+            return self._settings
+        else:
+            LOG.error('Skill not fully initialized. Move code ' +
+                      'from  __init__() to initialize() to correct this.')
+            LOG.error(simple_trace(traceback.format_stack()))
+            raise Exception('Accessed self.settings in __init__')
```

Comment:
Could this more precisely be a `RuntimeError`? Or some other more specific Exception?

Reply:
its the same exception used in other properties that do the same, for consistency lets keep them all the same

a dedicated exception could be added in a future PR